connector/extensions/web-service/src/main/java/org/oaebudt/edc/web/service/ReportService.java

```diff
@@ -0,0 +1,169 @@
+package org.oaebudt.edc.web.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonReader;
+import org.eclipse.edc.connector.controlplane.asset.spi.domain.Asset;
+import org.eclipse.edc.connector.controlplane.contract.spi.types.offer.ContractDefinition;
+import org.eclipse.edc.connector.controlplane.services.spi.asset.AssetService;
+import org.eclipse.edc.connector.controlplane.services.spi.contractdefinition.ContractDefinitionService;
+import org.eclipse.edc.connector.dataplane.http.spi.HttpDataAddress;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.query.Criterion;
+import org.eclipse.edc.spi.result.ServiceFailure;
+import org.eclipse.edc.spi.types.domain.DataAddress;
+import org.oaebudt.edc.web.dto.CreateAssetRequest;
+import org.oaebudt.edc.web.model.ReportType;
+import org.oaebudt.edc.web.repository.ReportStore;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+import static org.eclipse.edc.spi.constants.CoreConstants.EDC_NAMESPACE;
+
+public class ReportService {
+    private final ReportStore reportStore;
+    private final Monitor monitor;
+    private final AssetService assetService;
+    private final ContractDefinitionService contractDefinitionService;
+    private final URI consumerApiBaseUrl;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public ReportService(Monitor monitor, ReportStore reportStore, AssetService assetService,
+                         ContractDefinitionService contractDefinitionService, URI consumerApiBaseUrl) {
+        this.monitor = monitor;
+        this.reportStore = reportStore;
+        this.assetService = assetService;
+        this.contractDefinitionService = contractDefinitionService;
+        this.consumerApiBaseUrl = consumerApiBaseUrl;
+    }
+
+    public void createAsset(CreateAssetRequest request) {
+        validateCreateAsset(request);
+        MetadataValidator.validate(request.metadata());
+
+        createAssetInConnector(
+                request.title(),
+                request.metadata(),
+                request.reportType(),
+                request.accessDefinition(),
+                request.url(),
+                request.method(),
+                request.authHeaderKey(),
+                request.authCode(),
+                request.headers()
+        );
+    }
+
+    public void uploadAndCreateAsset(InputStream file, String title, String accessDefinition, String metadataJson, ReportType reportType) throws JsonProcessingException {
```

Comment:
service methods should return `ServiceResult`, so the caller can know what happened without blindly catch exceptions.

Reply:
Resolved